test/dates/rounding.jl

```diff
@@ -144,3 +144,69 @@ end
         end
     end
 end
+@testset "Rounding for periods" begin
+    x = Dates.Nanosecond(172799999999999)
+    @test floor(x, Dates.Week) == Dates.Week(0)
+    @test floor(x, Dates.Day) == Dates.Day(1)
+    @test floor(x, Dates.Hour) == Dates.Hour(47)
+    @test floor(x, Dates.Minute) == Dates.Minute(2879)
+    @test floor(x, Dates.Second) == Dates.Second(172799)
+    @test floor(x, Dates.Millisecond) == Dates.Millisecond(172799999)
+    @test floor(x, Dates.Microsecond) == Dates.Microsecond(172799999999)
+    @test floor(x, Dates.Nanosecond) == x
+    @test ceil(x, Dates.Week) == Dates.Week(1)
+    @test ceil(x, Dates.Day) == Dates.Day(2)
+    @test ceil(x, Dates.Hour) == Dates.Hour(48)
+    @test ceil(x, Dates.Minute) == Dates.Minute(2880)
+    @test ceil(x, Dates.Second) == Dates.Second(172800)
+    @test ceil(x, Dates.Millisecond) == Dates.Millisecond(172800000)
+    @test ceil(x, Dates.Microsecond) == Dates.Microsecond(172800000000)
+    @test ceil(x, Dates.Nanosecond) == x
+    @test round(x, Dates.Week) == Dates.Week(0)
+    @test round(x, Dates.Day) == Dates.Day(2)
+    @test round(x, Dates.Hour) == Dates.Hour(48)
+    @test round(x, Dates.Minute) == Dates.Minute(2880)
+    @test round(x, Dates.Second) == Dates.Second(172800)
+    @test round(x, Dates.Millisecond) == Dates.Millisecond(172800000)
+    @test round(x, Dates.Microsecond) == Dates.Microsecond(172800000000)
+    @test round(x, Dates.Nanosecond) == x
+end
+@testset "Rounding for periods that should not need rounding" begin
+    for x in [Dates.Week(3), Dates.Day(14), Dates.Microsecond(604800000000)]
+        local dt
+        for p in [Dates.Week, Dates.Day, Dates.Hour, Dates.Second, Dates.Millisecond, Dates.Microsecond, Dates.Nanosecond]
+            local p
```

Comment:
Unneeded?

Reply:
@kshyatt can you elaborate on why this is needed?

Reply:
This is a case where I noticed that `local dt` had been added to the tests in several places, so I replicated that syntax for these tests (in this case `local dt` should actually be `local x` above). But yeah, I don't actually know what those lines are meant to accomplish.